### PR TITLE
feat: expose content of static files via endpoint

### LINKED ISSUE #4: feat: Serve lists of flagged proposals and spaces
Currently, we are maintaining three lists for flagged proposals and spaces. To improve accessibility and integration, we need to make these lists available on the hub. To achieve this, we should move the lists to Sidekick repo and create an endpoints that serve them.

The existing lists can be found at the following locations:

For proposals (located on the 'Ui' repository):

1. [useProposals.ts#L5](https://github.com/snapshot-labs/snapshot/blob/ad9cb607fb8c33ee4dd86465a9f4361021856486/src/composables/useProposals.ts#L5)
1. [useProposals.ts#L45](https://github.com/snapshot-labs/snapshot/blob/ad9cb607fb8c33ee4dd86465a9f4361021856486/src/composables/useProposals.ts#L45)

For spaces (located in the 'snapshot-spaces' repository), including verified spaces:

1. [verified.json](https://github.com/snapshot-labs/snapshot-spaces/blob/67b67de03d39140e3fb0fd56215a65383006ba63/spaces/verified.json)

### PR DESCRIPTION
**Pull request by wa0x6e**
fix #4 

Serve the content of given .json files via the rest `/moderationList` API endpoint

Files are located inside the /data directory, and split in 3 different files for easier history tracking:

- flaggedLinks.json
- flaggedProposalIds.json
- verifiedSpaces.json

The endpoint also takes a `?fields=` query params, in order to select only the chosen list (default to all if not set) for a lighter request, e.g. `http://localhost:3000/moderationList?fields=verifiedSpaces,flaggedLinks,flaggedProposalIds`

### CODE REVIEW

**Comment by samuveth** [comment]
I bit hard to review since it includes changes from other PRs

**Comment by wa0x6e** [comment]
Yes, this PR is waiting for the first commit of the repo, installing express and other basic packages, to be pulled first.

Will ping you when it'll be ready.

For now, are the features listed in the description what you're expecting ? Meaning we still need to commit changes to files in the repo to make changes, instead of using a DB ?

**Comment by wa0x6e** [comment]
@samuveth Hey, PR is now clean of unrelated changes. There's only 2 main files for this feature, all the other are related to the tests.

For the .json files themselves, should we move them to another repo, only dedicated to serve json files (as I get that we may have more json files, with changes requested by user, and not team), so we could separate issues/tickets related to json files, apart from code ? (we still will use this repo to read the json file from here).

File: src/lib/moderationList.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import { readFileSync } from 'fs';
+import path from 'path';
+
+const FIELDS = ['flaggedLinks', 'flaggedProposalIds', 'verifiedSpaces'];
+const CACHE_PATH = path.resolve(__dirname, `../../${process.env.MODERATION_LIST_PATH || 'data'}`);
```
Comment (ChaituVR) [inline src/lib/moderationList.ts:5]: Can default it to `data` no need for env :) or we can have it as a variable or config.json if required, should have less env as possible, in this case, I don't think we will ever change the folder name 😅

File: src/lib/moderationList.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { readFileSync } from 'fs';
+import path from 'path';
+
+const FIELDS = ['flaggedLinks', 'flaggedProposalIds', 'verifiedSpaces'];
```
Comment (ChaituVR) [inline src/lib/moderationList.ts:4]: maybe can be just `flaggedProposals`

**Review by ChaituVR — CHANGES_REQUESTED** [review]

File: src/lib/moderationList.ts (hunk 1 above)
Comment (wa0x6e) [reply src/lib/moderationList.ts:5]: `MODERATION_LIST_PATH` is used by the test suite, to avoid mocking

File: src/lib/moderationList.ts (hunk 2 above)
Comment (wa0x6e) [reply src/lib/moderationList.ts:4]: We can, if we're sure that we will not have any other types of flagged proposal type in the future

File: src/lib/moderationList.ts (hunk 1 above)
Comment (ChaituVR) [reply src/lib/moderationList.ts:5]: maybe let's just use the relative path, i mean `data` folder inside `test` folder

File: src/lib/moderationList.ts (hunk 2 above)
Comment (ChaituVR) [reply src/lib/moderationList.ts:4]: we can add a suffix for future ones, for ids we can call them just proposals

File: package.json [hunk 3]
```diff
@@ -25,6 +27,7 @@
     "@aws-sdk/client-s3": "^3.316.0",
     "compression": "^1.7.4",
     "cors": "^2.8.5",
+    "dotenv-cli": "^7.2.1",
```
Comment (ChaituVR) [inline package.json:30]: I feel like better to move this one to dev

**Review by ChaituVR — APPROVED** [review]
Everything else looks good, I think we should actually add data :) so UI can test, and we can update the data later once everything is ready in UI and stop updating snapshot-spaces

**Comment by samuveth** [comment]
@wa0x6e please add the latest data from the original files so we can test/merge

**Comment by wa0x6e** [comment]
> @wa0x6e please add the latest data from the original files so we can test/merge

Done

**Comment by wa0x6e** [comment]
> Everything else looks good, I think we should actually add data :) so UI can test, and we can update the data later once everything is ready in UI and stop updating snapshot-spaces

What's the relation with snapshots-spaces ?

**Comment by samuveth** [comment]
Please add the flagged proposal ID we added today as well https://github.com/snapshot-labs/snapshot/commit/106650215d9340e27d2acf93ed16a12315ac50c0

Is this deployed now, where can we test?

**Comment by wa0x6e** [comment]
> Please add the flagged proposal IDs we added today as well [snapshot-labs/snapshot@1066502](https://github.com/snapshot-labs/snapshot/commit/106650215d9340e27d2acf93ed16a12315ac50c0)
> 
> Is this deployed now, where can we test?

Let's just go with that, until deployed and ready, and you will be free to submit PR to update these files

File: src/api.ts [hunk 4]
```diff
@@ -74,4 +75,15 @@ router.post('/votes/:id', async (req, res) => {
   }
 });
 
+router.get('/moderationList', async (req, res) => {
+  const { fields } = req.query;
+
+  try {
+    res.send(getModerationList(fields ? (fields as string).split(',') : undefined));
```
Comment (ChaituVR) [inline src/api.ts:82]: Should be `.json` so client knows this is a JSON

**Review by ChaituVR — CHANGES_REQUESTED** [review]
Everything else looks great and working well

**Review by ChaituVR — APPROVED** [review]
TestAck

**Pull request merged**